lib/dry/struct/setters.rb

```diff
@@ -39,6 +40,12 @@ def self.define_setter_for(struct:, attribute:, type:)
           end
         end
       end
+
+      def self.remove_trailling_question_mark(attribute)
+        return if attribute.class != Symbol
+        attribute = attribute.to_s.chomp "?"
+        return attribute.to_sym
+      end
```

Comment:
Any reason to care about `attribute` being a `Symbol` What about this?

```ruby
def self.remove_trailing_question_mark(attribute)
  attribute.to_s.chomp('?')
end
```

Having that, I probably wouldn't extract this into a method and just do this:

```ruby
def self.define_setter_for(struct:, attribute:, type:)
  attribute = attribute.to_s.chomp('?')
  setter = "#{attribute}=".to_sym

  struct.class_eval do
    unless instance_methods.include?(setter)
      define_method(setter) do |value|
        @attributes[attribute] = type.call(value)
      end

      setter
    end
  end
end
```

What do you think?

Reply:
Returning if the attribute isn't a symbol was to handle a separate thing ... I might raise as an issue to better explain that "thing", but will remove from here.

Reply:
I extracted it into its own method for readability. Seemed like the reasoning behind `chomp(?)` would be difficult to work out without some awareness of the issue. I've added a link to the issue for future readers of the code.